octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js

```diff
@@ -342,62 +286,48 @@ $(function () {
         };
 
         self.setDialog = function () {
-            if (self.is_dialog_open()) {
+            if (self.show_dialog()) {
                 self.showDialog();
-            } else {
+            } else if (!self.show_dialog()) {
                 self.hideDialog();
+            } else {
+                self._debugDaShit("setDialog() nothing to do");
             }
         };
 
         self.showDialog = function (force) {
             self._debugDaShit("showDialog() " + (force ? "force!" : ""));
-            self.dialogShouldBeOpen = true;
-            if (
-                (!self.dialogIsInTransition &&
-                    !self.dialogElement.hasClass("in")) ||
-                force
-            ) {
-                var param = "show";
-                if (!self.is_rtl_mode()) {
-                    // not dismissible in paused mode
-                    param = {
-                        backdrop: "static",
-                        keyboard: MRBEAM_ENV_LOCAL == "DEV",
-                    };
-                }
 
-                self._debugDaShit("showDialog() dialogIsInTransition <= true");
-                self.dialogIsInTransition = true;
-                self._debugDaShit("showDialog() show");
-                self.dialogElement.modal(param);
-
-                self._setTimeoutForDialog();
-            } else {
-                self._debugDaShit("showDialog() skip");
+            var param = "show";
+            if (self.is_pause_mode()) {
+                // not dismissible in paused mode
+                param = {
+                    backdrop: "static",
+                    keyboard: MRBEAM_ENV_LOCAL === "DEV",
+                };
             }
+
+            self._debugDaShit("showDialog() dialogIsInTransition <= true");
+            self._debugDaShit("showDialog() show");
+            self.dialogElement.modal(param);
         };
 
         self.hideDialog = function (force) {
             self._debugDaShit("hideDialog() " + (force ? "force!" : ""));
-            self.dialogShouldBeOpen = false;
-            if (
-                (!self.dialogIsInTransition &&
-                    self.dialogElement.hasClass("in")) ||
-                force
-            ) {
-                self._debugDaShit("hideDialog() dialogIsInTransition <= true");
-                self.dialogIsInTransition = true;
-                self._debugDaShit("hideDialog() hide");
-                self.dialogElement.modal("hide");
+            self._debugDaShit("hideDialog() hide");
+            self.dialogElement.modal("hide");
```

Comment:
We don't need this whole method; we can just use: self.dialogElement.modal("hide");

Reply:
changed